### PR TITLE
perf(sdk): parse and set session storage in login success callback

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

**Review by blackchoey — DISMISSED** [review]

File: packages/sdk/src/credential/teamsUserCredential.browser.ts [hunk 1]
```diff
@@ -116,12 +116,23 @@ export class TeamsUserCredential implements TokenCredential {
...
+            } catch (error) {
+              // If can not parse result as Json, will NOT throw error since user may return other info in auth-end page.
+              // TODO: resolve the result.
+              const failedToParseResult = "Failed to parse result to Json.";
+              internalLogger.verbose(failedToParseResult);
+              resolve();
+            }
+
+            // If sessionStorage exists in result, set the values in current session storage.
+            if (resultJson.sessionStorage) {
```
Comment (SLdragon) [inline packages/sdk/src/credential/teamsUserCredential.browser.ts:131]: if resultJson.sessionStorage is empty, do we need to throw error or show error messages?

File: packages/sdk/src/credential/teamsUserCredential.browser.ts [hunk 2]
```diff
@@ -116,12 +116,23 @@ export class TeamsUserCredential implements TokenCredential {
...
-              resolve(accessToken);
-            } catch (error: any) {
-              reject(error);
+              resultJson = JSON.parse(result);
+            } catch (error) {
+              // If can not parse result as Json, will NOT throw error since user may return other info in auth-end page.
+              // TODO: resolve the result.
+              const failedToParseResult = "Failed to parse result to Json.";
+              internalLogger.verbose(failedToParseResult);
+              resolve();
```
Comment (SLdragon) [inline packages/sdk/src/credential/teamsUserCredential.browser.ts:127]: remove this line or return after this to avoid multiple resolve() called

Comment (KennethBWSong) [reply packages/sdk/src/credential/teamsUserCredential.browser.ts:127]: Fixed. Thx!

File: packages/sdk/src/credential/teamsUserCredential.browser.ts [hunk 3]
```diff
@@ -116,12 +116,24 @@ export class TeamsUserCredential implements TokenCredential {
...
+              return;
             }
+
+            // If sessionStorage exists in result, set the values in current session storage.
+            if (resultJson.sessionStorage) {
+              this.setSessionStorage(resultJson.sessionStorage);
```
Comment (tecton) [inline packages/sdk/src/credential/teamsUserCredential.browser.ts:133]: When will the data in session storage be used? What's the expected data returned from auth-end page?

File: packages/sdk/src/credential/teamsUserCredential.browser.ts [hunk 4]
```diff
@@ -116,12 +116,23 @@ export class TeamsUserCredential implements TokenCredential {
...
+            let resultJson: any = {};
             try {
-              const accessToken = parseAccessTokenFromAuthCodeTokenResponse(result);
-              resolve(accessToken);
-            } catch (error: any) {
-              reject(error);
+              resultJson = JSON.parse(result);
+            } catch (error) {
+              // If can not parse result as Json, will NOT throw error since user may return other info in auth-end page.
+              // TODO: resolve the result.
```
[the remark this answers is not in the document]
Comment (KennethBWSong) [reply packages/sdk/src/credential/teamsUserCredential.browser.ts:124]: No, we need further discussion on whether we should return the result to user.

**Review by SLdragon — APPROVED** [review]

File: packages/sdk/src/credential/teamsUserCredential.browser.ts (hunk 3 above)
Comment (tecton) [reply packages/sdk/src/credential/teamsUserCredential.browser.ts:133]: Offline sync: it's for msal and auth-end page will be updated later.

**Review by tecton — APPROVED** [review]

**Review by blackchoey — APPROVED** [review]

**Pull request merged**